### PR TITLE
Add audit log event proto for user auto merged event

### PR DESCRIPTION
**Pull request by cyinma**

### CODE REVIEW

File: proto/buf/alpha/audit/v1alpha1/event.proto [hunk 1]
```diff
@@ -343,6 +345,8 @@ message PayloadUserLoggedIn {}
 
 message PayloadUserLoggedOut {}
 
+message PayloadUserMerged {}
```
Comment (unmultimedio) [inline proto/buf/alpha/audit/v1alpha1/event.proto:348]: Could you share some context of what is the event about? Is a user merging with another user?

Comment (bufdev) [reply proto/buf/alpha/audit/v1alpha1/event.proto:348]: Let's get in the habit of adding documentation for any new type as well, even though the previous payload types dont have it.

```
// This message represents ... (3-5 sentences on what this is)
```

Comment (cyinma) [reply proto/buf/alpha/audit/v1alpha1/event.proto]: chatted offline with @unmultimedio , this audit event is for the case when a new user from a new IdP login, and we automatically merge it to an existing user from the old IdP. renamed the event and message to clarify a bit and added docs

Comment (unmultimedio) [reply proto/buf/alpha/audit/v1alpha1/event.proto]: When looking at this event, we'll see that another user from _some_ IdP got merged into, with no additional info... I know subject is probably not needed here, but what about at least the IdP provider id or name? So at least we know that for user John Doe we recognized and automerged his IdP user from his ID in whatever IdP provider he used?

Comment (cyinma) [reply proto/buf/alpha/audit/v1alpha1/event.proto]: I don't think we need to log those information? basically we wouldn't know which IdP created the existing user, and logging the current IdP doesn't seems very useful

**Review by bufdev — APPROVED** [review]

**Review by unmultimedio — APPROVED** [review]

**Review by saquibmian — APPROVED** [review]

**Pull request merged**